src/core_modules/capture-core/components/WidgetEnrollment/HOC/withOrganizationUnit.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+import React, { useMemo, type ComponentType } from 'react';
+import log from 'loglevel';
+import i18n from '@dhis2/d2-i18n';
+import { errorCreator } from 'capture-core-utils';
+import { useDataQuery } from '@dhis2/app-runtime';
+import { LoadingMaskElementCenter } from '../../LoadingMasks';
+
+type Props = {
+    ownerOrgUnit: string,
+};
+const ENROLLMENT_RETRIEVE_ERROR =
+    'Enrollment widget could not be loaded. Please try again later';
+
+export const withOrganizationUnit = (Component: ComponentType<any>) => (
+    props: Props,
+) => {
+    const { error, loading, data } = useDataQuery(
+        useMemo(
+            () => ({
+                organisationUnits: {
+                    resource: `organisationUnits/${props.ownerOrgUnit}`,
+                    params: {
+                        fields: ['displayName'],
+                    },
+                },
+            }),
+            [props.ownerOrgUnit],
+        ),
+    );
+
+    if (error) {
+        log.error(errorCreator(ENROLLMENT_RETRIEVE_ERROR)({ error }));
+        return <span>{i18n.t(ENROLLMENT_RETRIEVE_ERROR)} </span>;
```

Comment:
I wonder with this approach when `withOrganizationUnit` , `withProgram` and `withEnrollment` all have error at the same time it will return 3 times the same message error? 
Should we just display it once

Reply:
Hey @huocha,

The error message will only be displayed once. When an error happens in any of the HOCs then its child `Component` is not rendered anymore. The "second error" will never appear because the next calls to the API don't even happen.

Let me know if this explanation clears your doubts. Thank you!

Reply:
@simonadomnisoru thank you for your clear explanation :) that's really helpful. Great work!